File: diet-plan/java_api_dietmodel/Meal.java
import java.util.ArrayList;

class Meal {
    String name;
    double protein, calories;
    int proteinLevel, caloriesLevel;//1=low, 2=med, 3=high

    ArrayList<Ingredient> ingredientList;
    Meal(String mealName,  ArrayList<Ingredient> ListOfIngredient) {

        this.name = mealName;
        this.ingredientList = ListOfIngredient;

        protein = 0.0;
        calories = 0.0;

        setStats();
    }
    void setStats() {

        for(int i=0; i<ingredientList.size(); i++) {
            protein = protein+ ingredientList.get(i).getProtein();
            calories = calories+ ingredientList.get(i).getCalories();
        }

        if (protein<10)
            proteinLevel = 1;
        else if (protein < 35)
            proteinLevel = 2;
        else
            proteinLevel = 3;

        if (calories<250)
            caloriesLevel = 1;
        else if (calories < 450)
            caloriesLevel = 2;
        else
            caloriesLevel = 3;
    }

    public int getCaloriesLevel() {
        return caloriesLevel;
    }

    public int getProteinLevel() {
        return proteinLevel;
    }
}

File: Meal.js
class Meal {
    constructor(mealName, ListOfIngredient) {
      this.name = null;
      this.protein = 0;
      this.calories = 0;
      this.proteinLevel = 0;
      this.caloriesLevel = 0;
      this.ingredientList = [];
  
      this.name = mealName;
      this.ingredientList = ListOfIngredient;
      this.protein = 0.0;
      this.calories = 0.0;
      this.setStats();
    }
    setStats() {
      for (let i = 0; i < this.ingredientList.size(); i++) {
        this.protein = this.protein + this.ingredientList.get(i).getProtein();
        this.calories = this.calories + this.ingredientList.get(i).getCalories();
      }
      if (this.protein < 10) {
        this.proteinLevel = 1;
      } else {
        if (this.protein < 35) {
          this.proteinLevel = 2;
        } else {
          this.proteinLevel = 3;
        }
      }
      if (this.calories < 250) {
        this.caloriesLevel = 1;
      } else {
        if (this.calories < 450) {
          this.caloriesLevel = 2;
        } else {
          this.caloriesLevel = 3;
        }
      }
    }
    getCaloriesLevel() {
      return this.caloriesLevel;
    }
    getProteinLevel() {
      return this.proteinLevel;
    }
  }
  